Send job stats to keen

svenfuchs opened the pull request.

joshk commented on lib/travis/hub/serialize/keen/job.rb:
```diff
@@ -0,0 +1,92 @@
+                  repository_slug:    repo.slug,
+                  repository_private: repo.private,
+                  build_id:           job.source_id,
+                  commit_id:          job.commit_id,
```
i don't think we need this

joshk commented on lib/travis/hub/serialize/keen/job.rb:
```diff
@@ -0,0 +1,92 @@
+              jobs: [
+                {
+                  id:                 job.id,
+                  repository_id:      repo.id,
```
could we also add:
`owner_type`
`owner_id`
`owner_name`
`repository_name`

I think we can also remove the `repository_slug` field.

svenfuchs commented:
@joshk I have now:

* kept `repository_slug` as i think that's more relevant than just `repository_name`?
* kept `repository_private` as i think that might be relevant for some queries?
* added `owner_id`, `owner_type`, and `owner_login` (instead of `owner_name`, as that's what we usually use to refer to an owner internally)
* removed `build_id` and `commit_id`
* added `created_at`, `received_at`, `started_at`, `queued_at`, `canceled_at`, `finished_at`

WDYT?

roidrage commented:
I'd love to see language and language version as well, which would allow us to run better statistics on those attributes than we can right now.

Does this send a job payload to Keen for every state change of a single job or just when it's finished?

joshk commented:
I think it would be good to add `build_id` back as we can do counts on jobs with unique build ids.

As for `created_at`, `received_at`, `started_at`, `queued_at`, `canceled_at`, `finished_at`, I'm not sure if keen can do calculations on this data or if it would be better for us to send summary info. I'm thinking it might be better to send info like `waiting_time`, `queued_time`, `booting_time`, `running_time`.

Also, lets add the language and language versions for now too, as this will give us a more true and correct understand of what is running on travis.

Thoughts?

roidrage commented:
I don't think Keen can do calculations over several fields. It can e.g. sum on a single field. So storing durations would indeed serve us better here. That way we could also start pulling these metrics for our customers and give them trends.

A few more things come to mind for me that would be handy:
* Differentiate between pull request and push jobs as well as api.
* Differentiate public and private repos. This would allow us to store both .org and .com in the same Keen project, giving us more possibilities to aggregate the data.
* Queue, operating system and those kinds of flags. That we could get a much better overview of how jobs are spread out across our infrastructures.
* Storing specific information about commits, authors, committers, and so on. This would allow us to determine unique authors and committers using our platform.

I'm also :+1: on bringing back `build_id` for the same reason @joshk mentioned. Maybe even store the build number? That could be handy to determine how many new projects we've added in a week.

roidrage commented:
Just linkifying this to https://github.com/travis-pro/post-its/issues/113 as well, as that's where I'd posted most of the questions relevant for tracking more job data.
